src/Fieldset/index.jsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import newId from '../utils/newId';
+import styles from './Fieldset.scss';
+import ValidationMessage from '../ValidationMessage/index';
+import Variant from '../utils/constants';
+
+const inputProps = {
+  variantIconDescription: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+  children: PropTypes.node,
+  className: PropTypes.arrayOf(PropTypes.string),
+  id: PropTypes.string,
+  isValid: PropTypes.bool,
+  legend: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
+  invalidMessage: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+  variant: PropTypes.shape({
+    status: PropTypes.string,
+  }),
+};
+
+const defaultProps = {
+  variantIconDescription: '',
+  children: null,
+  className: [],
+  id: '',
+  isValid: true,
+  invalidMessage: '',
+  variant: {
+    status: Variant.status.INFO,
+  },
+};
+
+class Fieldset extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const id = props.id || newId('fieldset');
+    this.state = {
+      id,
+    };
+  }
+
+  componentWillReceiveProps(nextProps) {
+    // TODO: use getDerivedStateFromProps when upgraded to React 16.3+
+    if (nextProps.id !== this.props.id) {
+      if (nextProps.id) {
+        this.setState({
+          id: nextProps.id,
+        });
+      } else {
+        this.setState({
+          id: newId('fieldset'),
+        });
+      }
+    }
+  }
+
+  getVariantFieldsetClassName() {
+    const { variant } = this.props;
```

Comment:
Can this be simplified to

```javascript
if (this.props.variant.status === Variant.status.INFO) {
  return styles['is-invalid-nodanger'];
}
```

Reply:
That is simpler, but the current structure makes it really easy to add another variant.

Reply:
Fair - I guess shouldn't that be something the implementer that wants to add more variants should solve? Especially since it's unclear if more variants will be needed (something something pre-mature optimization)?

Either way, not a big deal.